keps/sig-api-machinery/3157-watch-list/README.md

```diff
@@ -797,23 +802,29 @@ Pick one of these and delete the rest.
 
 - [x] Feature gate (also fill in values in `kep.yaml`)
   - Feature gate name: WatchList
-  - Components depending on the feature gate: the kube-apiserver
+  - Components depending on the feature gate: 
+    - kube-apiserver
+    - kube-controller-manager
 - [ ] Other
   - Describe the mechanism:
   - Will enabling / disabling the feature require downtime of the control
-    plane?
+    plane? 
+Yes.
+After graduation to beta the feature will be automatically enabled for kube-apiserver and kube-controller-manager.
+Disabling requires passing `WatchList=false` to `feature-gates` command line flag for both components.
   - Will enabling / disabling the feature require downtime or reprovisioning
-    of a node?
+    of a node? No.
 
 ###### Does enabling the feature change any default behavior?
-No.
+No. User have to explicitly set `ENABLE_CLIENT_GO_WATCH_LIST_ALPHA` environment variable.
 <!--
 Any change of default behavior may be surprising to users or break existing
 automations, so be extremely careful here.
 -->
 
 ###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
-Yes, in that scenario the kube-apiserver will reject WATCH requests with the new query parameter forcing informers to fall back to the previous mode.
+Yes, by passing `WatchList=false` to `feature-gates` command line flag to the components for which this feature was enabled. 
+Note that the kube-apiserver will reject WATCH requests with the new query parameter forcing informers to fall back to the previous mode.
```

Comment:
Will the fallback happen for informers that were started against a server that honors WatchList?

Reply:
In general yes. 
But this largely depends on the component for which this feature was enabled. For example  KCM will be restarted because the leader election lock will be lost.

Reply:
I don't understand your answer - how is leader-election related here?
Leader-election doesn't use neither/list watch, so that is completely irrelevant here.


Getting back to David's question - the reflector, whenever it needs to perform a LIST operation (if the option is enabled) it always tries to do list-watch:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/cache/reflector.go#L330
and if that fails, falls back to list+watch

So this should work fine for both enablement & disablement of the FG without any restart.

Reply:
based on past experience, I recommend updating to always fallback to regular list no matter what error happened.

Reply:
I'm fine with falling back on any error.

Reply:
> based on past experience, I recommend updating to always fallback to regular list no matter what error happened.

Sure - that sounds reasonable

Reply:
clarify in doc to resolve thread.